subprojects/docs/src/docs/userguide/upgrade_version_4.adoc

```diff
@@ -129,6 +129,11 @@ For now, by-convention config files in subprojects will override any in the root
  * `getEnabledDirectoryReportDestinations()`, `getEnabledFileReportDestinations()` and `getEnabledReportNames()` have all been removed from `org.gradle.api.reporting.ReportContainer`.
  * link:{javadocPath}/org/gradle/StartParameter.html#getProjectProperties--[StartParameter.projectProperties] and link:{javadocPath}/org/gradle/StartParameter.html#getSystemPropertiesArgs--[StartParameter.systemPropertiesArgs] now return immutable maps.
 
+[[changes_5.0]]
+== Changes in 5.0
+
+* <<#rel5.0:pom_optional_dependencies,Maven optional dependencies are no longer interpreted as dependency constraints>>
```

Comment:
I would rephrase this to explain it was a feature available behind the `IMPROVED_POM_SUPPORT` otherwise it looks scary. A bit like you did in the other section.

Reply:
Or it might be left for later, when we mention here that `IMPROVED_POM_SUPPORT` became the default and list the promoted features.

Reply:
Yeah I had the same thought. I just wanted to have a minimal update of the docs here, knowing that we're going to enable improved POM support by default at some point. In which case I would conflate the 2 sections into one, and just say that as part of this work we abandoned the promotion of optional dependencies to constraints.

Reply:
👍 Works for me